code/script.js

```diff
@@ -222,104 +228,152 @@ const generateBoard = () => {
-  // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
-
+  const value = questions.value
+  // These variables store what option group (category) and value the question (option) belongs to.
   currentQuestion = {
     category: category,
-    // value: value
+    value: value
   }
 }
 
-// This function should be invoked when you click on 'Find Out' button.
+//CHECK QUESTION //
+//This function should be invoked when you click on 'Find Out' button.
 const checkQuestion = () => {
   const { category, value } = currentQuestion
 
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
+  // COMPARE THE currentQuestion WITH THE SECRET PERSON based on category (hair, etc). See if we should keep or remove people based on that
+  //Spent some on a mistake made here because when hair and eyes were correctly guessed, it first showed that the answer was right - then that it was wrong - and then the wholeboard disappeared. Turned out to be another curly bracket issue.. 
   if (category === 'hair' || category === 'eyes') {
+    console.log(category, value)
+    if (secret[category] === value) {
+      filterCharacters(true)
+    } else {
+      filterCharacters(false)
+    }
+  }
 
-  } else if (category === 'accessories' || category === 'other') {
-
+  else if (category === 'accessories' || category === 'other') {
+    console.log(category, value)
+    if (secret[category].includes(value)) {
+      filterCharacters(true)
+    }
+    else {
+      filterCharacters(false)
+    }
   }
 }
 
-// It'll filter the characters array and redraw the game board.
+
+// FILTER THE CHARACHTERS AND REDRAW BOARD GAME //
+//Alert messages for every fallout
 const filterCharacters = (keep) => {
   const { category, value } = currentQuestion
-  // Show the correct alert message for different categories
   if (category === 'accessories') {
     if (keep) {
       alert(
-        `Yes, the person wears ${value}! Keep all people that wears ${value}`
+        `Yes, the person wears ${value}! Keep all people who wear ${value}`
       )
     } else {
       alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
+        `No, the person doesn't wear ${value}! Remove all people who wear ${value}`
       )
     }
   } else if (category === 'other') {
-    // Similar to the one above
-  } else {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+      alert(`Yes, the person wears ${value}! Keep all people who wear ${value}`
+      )
+    } else {
+      alert(
+        `No, the person doesn´t wear ${value}! Remove all people who wear ${value}`
+      )
+    }
+  } else if (category === 'hair') {
```

Comment:
the readability of your filterCharacters function can be improved by avoiding the repetition of alert messages. You can create a single alert message and use template literals to make the code cleaner.

like the following : 

const filterCharacters = (keep) => {
  const { category, value } = currentQuestion;
  let message;

  if (category === 'accessories') {
    message = `person wears ${value}`;
  } else if (category === 'other') {
    message = `person wears ${value}`;
  } else if (category === 'hair') {
    message = `person has ${value} hair`;
  } else if (category === 'eyes') {
    message = `person has ${value} eyes`;
  }

  if (keep) {
    alert(`Yes, the ${message}! Keep all people who ${message}`);
  } else {
    alert(`No, the ${message}! Remove all people who ${message}`);
  }

  // FILTER BY CATEGORY TO KEEP/REMOVE BASED ON THE KEEP VARIABLE//
  charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value) === keep);

  // Invoke a function to redraw the board with the remaining people.
  generateBoard();
};